### ISSUE TITLE
Use `throw render()` / `throw redirect()` inside React/Vue/Solid comopnents

### ISSUE THREAD

**Issue by brillout** [vikejs/vike#1707]
### Description

In principle, it should be possible to use `throw render()` and `throw redirect()` inside UI components.

For this to work, `vike-{react,vue,solid}` need to catch the exception and then re-throw it inside the `onRenderHtml()`/`onRenderClient()` hook, so that the exception propagates to Vike.

Contribution welcome.

**Comment by brillout** [comment]
Deprioritizing, as per https://github.com/vikejs/vike-react/pull/123#issuecomment-2196236446.

**Comment by simplecommerce** [comment]
Is this not supposed to work already?
In my test project, I implemented `vike-node` and am using throw redirect() in a component I have, but only in SSR in order to redirect to another page when the user isn't logged-in, and it seems to work fine.

**Comment by brillout** [comment]
@simplecommerce Yes it already works if the `onRenderHtml()` and `onRenderClient()` hooks receive the exception thrown by `throw redirect()` / `throw render()`, and if you aren't using HTML streaming. Because it isn't clear what should happen when `throw redirect()` / `throw render()` is called during HTML Streaming (will browsers respect the redirect header after the HTML stream has started?), I'd recommend to define user redirections on a page level instead of a component level.
